includes/compat/class-wc-stripe-subscriptions-helper.php

```diff
@@ -24,23 +31,33 @@ public static function is_subscriptions_enabled() {
 	}
 
 	/**
-	 * Loads up to 50 subscriptions, and attempts to return up to 5 of those that are detached from the customer.
+	 * Loads up to 50 subscriptions, and attempts to return those that are detached from the customer.
 	 *
 	 * @return array
 	 *
 	 * @deprecated 9.6.0 This method is no longer used and will be removed in a future version.
 	 */
 	public static function get_some_detached_subscriptions() {
 		_deprecated_function( __METHOD__, '9.6.0' );
+		return self::get_detached_subscriptions( 50 );
+	}
+
+	/**
+	 * Loads all subscriptions, and attempts to return those that are detached from the customer.
+	 *
+	 * @param int $limit The maximum number of subscriptions to retrieve. Use -1 for no limit (default).
+	 * @return array
+	 */
+	public static function get_detached_subscriptions( $limit = -1 ) {
 		// Check if we have a cached result.
 		$cached_subscriptions = get_transient( self::DETACHED_SUBSCRIPTIONS_TRANSIENT_KEY );
-		if ( ! empty( $cached_subscriptions ) ) {
+		if ( is_array( $cached_subscriptions ) ) {
```

Comment:
As @annemirasol pointed out, this should fix the lack of caching for an empty list.